modules/videoio/src/cap_avfoundation_mac.mm

```diff
@@ -293,7 +293,7 @@ - (IplImage*)getOutput;
 
     [mCaptureSession release];
     [mCaptureDeviceInput release];
-    [mCaptureDevice release];
+    // [mCaptureDevice release]; fix #7833
```

Comment:
Probably we need `mCaptureDevice = nil;` here at least.

And the same fix [here](https://github.com/opencv/opencv/blob/3.2.0/modules/videoio/src/cap_avfoundation.mm#L306).